packages/@aws-cdk/aws-s3/lib/bucket.ts

```diff
@@ -1692,6 +1719,24 @@ export class Bucket extends BucketBase {
       };
     });
   }
+
+  private getOrCreateAutoDeleteObjectsResource() {
+    return CustomResourceProvider.getOrCreate(this, AUTO_DELETE_OBJECTS_RESOURCE_TYPE, {
+      codeDirectory: path.join(__dirname, 'auto-delete-objects-handler'),
+      runtime: CustomResourceProviderRuntime.NODEJS_12,
+      policyStatements: [
+        {
+          Effect: 'Allow',
+          Resource: this.bucketArn,
+          Action: [
+            // TODO: add key perms?
```

Comment:
Question on the TODO, will the custom resource be able to delete objects encrypted by KMS? I think we can give the same permissions as configured in `grantReadWrite` helper on the basis of encryption configuration.

Reply:
> will the custom resource be able to delete objects encrypted by KMS? I think we can give the same permissions as configured in grantReadWrite helper on the basis of encryption configuration.

That's the goal! Yeah I think that's probably what I will end up doing.

> If resource returned by CustomResourceProvider implements IGrantable, we can directly use the above mentioned helper

Hmm... so when I looked into this approach before, I wasn't able to find a way to get this to work, because I don't think `CustomResource` implements `IGrantable` (and I couldn't see a way to create a dummy/wrapper class to work around this). Likewise, `CustomResourceProvider.getOrCreate` just returns a service token string unfortunately, so it's not currently possible to get access to the role that it creates (which is the principal we would hypothetically be performing grant operations on). I'm definitely open to suggestions or ideas though.

Reply:
@Chriscbr you can pass [policies](https://github.com/aws/aws-cdk/blob/master/packages/@aws-cdk/core/lib/custom-resource-provider/custom-resource-provider.ts#L47) to the provider that will be added to its role. Will that be suffice?

Reply:
I believe that's the prop I'm currently passing the policies into - please clarify if you had something else in mind.